### PR TITLE
Add command docs

### PR DESCRIPTION
**Pull request by asmello**
Also sneakily makes some small tweaks to the init command error and variable naming.

### CODE REVIEW

**Comment by asmello** [comment]
Fixes #44

File: docs/src/commands/buffrs-generate.md [hunk 1]
```diff
@@ -1 +1,13 @@
...
+
+# Synopsis
+
+`buffrs generate --lang <LANGUAGE>`
+
+# Description
+
+This command uses a protocol buffer compiler to transform the `.proto` files under the `proto/` subtree to auto-generated source code. 
+
+Currently the only supported value for the language is `rust`.
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-generate.md:13]:
```suggestion
Currently the only supported value for the language is `python`.
```

File: docs/src/commands/buffrs-help.md [hunk 2]
```diff
@@ -1 +1,7 @@
 # buffrs help
+
+Prints out information about how to use the CLI.
+
+# Synopsis
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-help.md:5]:
```suggestion
### Synopsis
```

File: docs/src/commands/buffrs-help.md [hunk 3]
```diff
@@ -1 +1,7 @@
 # buffrs help
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-help.md:1]:
```suggestion
## buffrs help
```

Comment (mara-schulke) [reply docs/src/commands/buffrs-help.md:1]: this is to keep the overall structure in the book correct

File: docs/src/commands/buffrs-login.md [hunk 4]
```diff
@@ -1 +1,22 @@
 # buffrs login
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-login.md:1]: (applies to all pages)

File: docs/src/commands/buffrs-login.md [hunk 5]
```diff
@@ -1 +1,22 @@
...
+
+Saves an authentication token in the credentials store.
+
+# Synopsis
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-login.md:5]:
```suggestion
### Synopsis
```

**Comment by asmello** [comment]
- [ ] Add indices (and maybe brief intro) to command section pages.

File: docs/src/commands/buffrs-login.md [hunk 6]
```diff
@@ -1 +1,23 @@
...
+The token is currently stored in `$HOME/.buffrs/credentials.toml` in the
+following format:
+
+```toml
+[[credentials]]
+uri = "https://example.com/artifactory"
+token = "<secret>"
+```
+
+In the future this may change to use system native keychains.
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-login.md:23]: Well yes and no, it was a native system keychain and we had to move away :/

Comment (asmello) [reply docs/src/commands/buffrs-login.md:23]: Oh, curious as to why, but I'll remote that remark then.

File: docs/src/commands/buffrs-publish.md [hunk 7]
```diff
@@ -1 +1,40 @@
...
+
+In order for this command to be successful, the registry must be reachable via
+the network, and if authorization is required, credentials must have been
+previously saved via a [`buffrs login`](buffrs-login.md) invocation.
+
+By default, Buffrs does not allow publishing packages from git repositories in a
+dirty state (note: this requires the `git` feature to be enabled). This
+behaviour can be overridden by passing the `--allow-dirty` flag.
+
+#### Supported package types
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-publish.md:31]:
```suggestion
#### Supported project types
```

File: docs/src/commands/buffrs-uninstall.md [hunk 8]
```diff
@@ -1 +1,22 @@
-# buffrs uninstall
+## buffrs uninstall
+
+Deletes all dependencies from the local store.
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-uninstall.md:3]:
```suggestion
Deletes all installed dependencies from the local filesystem.
```

File: docs/src/commands/buffrs-publish.md [hunk 9]
```diff
@@ -1 +1,40 @@
...
+
+Both library and API packages can be published -- the only exception is
+implementation packages, which are deemed to be terminal packages in the
+dependency graph. This may change in the future. More details in [Package
+Types](../guide/package-types.md).
```
[the remark this answers is not in the document]
Comment (asmello) [reply docs/src/commands/buffrs-publish.md:36]: Cool, SGTM.

File: docs/src/commands/buffrs-uninstall.md [hunk 10]
```diff
@@ -1 +1,22 @@
...
+
+### Synopsis
+
+`buffrs uninstall`
+
+### Description
+
+This command does the reverse operation from the
+[`install`](buffrs-uninstall.md) command, and will clear out the `proto/vendors`
+directory, thus removing all installed dependencies from the local storage. This
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-uninstall.md:13]:
```suggestion
directory, thus removing all installed dependencies from the local filesystem. This
```

File: docs/src/commands/buffrs-uninstall.md [hunk 11]
```diff
@@ -1 +1,22 @@
...
+is generally safe to do as the `vendors` directory is managed by Buffrs and
+shouldn't contain any custom proto files. Subsequently invoking the install
+command should restore the exact same files, assuming the lockfile hasn't been
+regenerated.
+
+Since Buffrs currently blindly extracts packages without checking if a previous
+version was already installed, it is good practice to run this command before
+attempting to upgrade dependencies, to make sure installation starts from a
+clean state.
```
Comment (mara-schulke) [inline docs/src/commands/buffrs-uninstall.md:22]: This shouldnt be the case, lets remove this paragraph and clear the local `proto/vendors` dir before installing?

File: docs/src/commands/build-commands.md [hunk 12]
```diff
@@ -1 +1,10 @@
 # Build Commands
+
+Build commands assist implementation projects with generating code and
+documentation from Buffrs-managed protocol buffer files.
+
+## Index
+
+* buffrs clean
```
Comment (mara-schulke) [inline docs/src/commands/build-commands.md:8]:
```suggestion
* [buffrs clean]()
```

File: docs/src/commands/build-commands.md [hunk 13]
```diff
@@ -1 +1,10 @@
...
+* buffrs doc
```
Comment (mara-schulke) [inline docs/src/commands/build-commands.md:9]:
```suggestion
* [buffrs doc]()
```

File: docs/src/commands/general-commands.md [hunk 14]
```diff
@@ -1 +1,11 @@
...
+
+General commands interface with the CLI itself, in order to obtain access to
+built-in [help](buffrs-help.md) or [version]()
+information.
+
+## Index
+
+* [buffrs](buffrs.md)
+* [buffrs help](buffrs-help.md)
+* buffrs version
```
Comment (mara-schulke) [inline docs/src/commands/general-commands.md:11]: same here

File: docs/src/commands/manifest-commands.md [hunk 15]
```diff
@@ -1 +1,15 @@
...
+These commands can be used to include, remove and update dependencies while
+keeping the lockfile in sync.
+
+## Index
+
+* [buffrs add](buffrs-add.md)
+* buffrs lock
+* [buffrs remove](buffrs-remove.md)
+* buffrs tree
+* buffrs update
```
Comment (mara-schulke) [inline docs/src/commands/manifest-commands.md:15]: and here

**Review by mara-schulke — APPROVED** [review]

**Comment by mara-schulke** [comment]
Left some comments but this is gold! Thank you @asmello!!!

File: docs/src/commands/build-commands.md (hunk 12 above)
Comment (asmello) [reply docs/src/commands/build-commands.md:8]: Is this really better? It gives off the false impression that the page exists, but the link actually points to the origin page.

Comment (mara-schulke) [reply docs/src/commands/build-commands.md:8]: If it doesnt, you can resolve this, at least in the main summary it does.

I would remove this line if this is not a link tho! Its potentially confusing and could be read as a bug in the docs

File: docs/src/commands/buffrs-uninstall.md (hunk 11 above)
Comment (asmello) [reply docs/src/commands/buffrs-uninstall.md:22]: I guess that's reasonable, if we make it super clear that buffrs is fully in charge of the vendors directory.

Comment (asmello) [reply docs/src/commands/buffrs-uninstall.md]: https://github.com/helsing-ai/buffrs/issues/100

Comment (mara-schulke) [reply docs/src/commands/buffrs-uninstall.md]: We could verify checksums before clearing or similar – lets think about how to warn peeps in a tracking issue / discussion 🌞

**Pull request merged**